test/e2e/upgrade_test.go

```diff
@@ -31,15 +31,69 @@ const (
 	windowsWorkloadTesterJob = "windows-workload-tester"
 )
 
+// getNodeTypeAffinities generate node affinity for non-BYOH and BYOH nodes if present
+func getNodeTypeAffinities() ([]*v1.Affinity, error) {
+	var affinities []*v1.Affinity
+	if gc.numberOfMachineNodes >= 1 {
+		affinityNoByoh, err := getNodeAffinityForLabel(v1.NodeSelectorOpDoesNotExist, controllers.BYOHLabel)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error creating node affinity without BYOH label")
+		}
+		affinities = append(affinities, affinityNoByoh)
+	}
+	if gc.numberOfBYOHNodes >= 1 {
+		affinityByoh, err := getNodeAffinityForLabel(v1.NodeSelectorOpIn, controllers.BYOHLabel, "true")
+		if err != nil {
+			return nil, errors.Wrapf(err, "error creating node affinity for BYOH label")
+		}
+		affinities = append(affinities, affinityByoh)
+	}
+	return affinities, nil
+}
+
+// generateLinuxWorkloadTesterCommand creates a Job to curl Windows webserver every 5 seconds.
+// If the webserver is not accessible for few minutes Curl Windows webserver
+// exit the pod with an error.
+func generateLinuxWorkloadTesterCommand(clusterIP string) []string {
```

Comment:
We don't want multiple ways calling the curler jobs. Please see [testWindowsExporter](https://github.com/openshift/windows-machine-config-operator/blob/master/test/e2e/metrics_test.go#L37). We ideally should have a single method that kicks of a curler job with different inputs.

Reply:
Created [WINC-677](https://issues.redhat.com/browse/WINC-677)